cpp/net/connection_pool.cc

```diff
@@ -113,29 +101,21 @@ int ConnectionPool::Connection::SSLVerifyCallback(const int preverify_ok,
 }
 
 
-ConnectionPool::Connection::Connection(evhtp_connection_t* conn,
-                                       HostPortPair&& other_end)
-    : conn_(CHECK_NOTNULL(conn)),
-      other_end_(move(other_end)),
-      ssl_(conn_->ssl) {
-  if (ssl_) {
-    {
-      lock_guard<mutex> lock(ssl_mutex_);
-      CHECK(connections_by_ssl_.insert(make_pair(ssl_, this)).second);
-    }
-    SSL_set_tlsext_host_name(ssl_, other_end_.first.c_str());
-  }
+// static
+int ConnectionPool::Connection::GetSSLConnectionIndex() {
+  static const int ssl_connection_index(
+      SSL_get_ex_new_index(0, nullptr, nullptr, nullptr, nullptr));
```

Comment:
Fairly sure this gets initialised the first time the that `GetSSLConnectionIndex()` is called, so it should be well after SSL is initialised (it's also threadsafe since C++0x.)